ntp-daemon/src/peer.rs

```diff
@@ -137,27 +135,30 @@ pub async fn start_peer<A: ToSocketAddrs + std::fmt::Debug, C: 'static + NtpCloc
                                 recv_timestamp,
                             );
 
-                            let system_poll = system_snapshot.poll_interval.as_duration();
-                            let accept = peer.accept_synchronization(
-                                ntp_instant,
-                                config.frequency_tolerance,
-                                config.distance_threshold,
-                                system_poll,
-                            );
-
-                            if accept.is_err() {
-                                let _ = tx.send(MsgForSystem::NoMeasurement);
-                            } else  {
-                                match result {
-                                    Ok(update) => {
+                            match result {
+                                Ok(update) => {
+                                    let system_poll = system_snapshot.poll_interval.as_duration();
+                                    let accept = peer.accept_synchronization(
+                                        ntp_instant,
+                                        config.frequency_tolerance,
+                                        config.distance_threshold,
+                                        system_poll,
+                                    );
+
+                                    if accept.is_err() {
+                                        let _ = tx.send(MsgForSystem::NoMeasurement);
+                                    } else  {
                                         let _ = tx.send(MsgForSystem::Snapshot(update));
                                     }
-                                    Err(IgnoreReason::KissDemobilize) => {
-                                        let _ = tx.send(MsgForSystem::MustDemobilize);
```

Comment:
this was unreachable before because a packet with a kiss code would never be accepeted